flycheck.el

```diff
@@ -3613,25 +3613,71 @@ show the icon."
 
 
 ;;; Built-in error levels
+(defun flycheck-high-dpi-screen ()
+  "Return t if the user has a HiDPI screen"
+  (let* ((attrs (car (display-monitor-attributes-list)))
+         (size (assoc 'mm-size attrs))
+         (sizex (cadr size))
+         (res (cdr (assoc 'geometry attrs)))
+         (resx (- (caddr res) (car res)))
+         dpi)
+    (catch nil
+      ;; if dpi > 150 (standard dpi: 96; hdpi: 192)
+      (if (> (* (/ (float resx) sizex) 25.4) 150) t nil))))
```

Comment:
This looks good! But I think it could be simplified by just using `x-display-mm-width` and `x-display-pixel-width`.

Reply:
I agree that using `(x-display-name)` and `(x-display-mm-width)` would be simpler.  However, the documentation for those function notes


```
On "multi-monitor" setups this refers to the width in millimeters for
all physical monitors associated with TERMINAL.  To get information
for each physical monitor, use ‘display-monitor-attributes-list’.
```

So, to support multi-monitor setups, I think it would be better to stick with `(display-monitor-attributes-list)`.  I added a comment to this effect.

Reply:
Thanks for pointing this out.  I emailed emacs-devel to ask about this.